Syscollector compilation in Windows is not working

# Windows compilation failure

## Description
The `dev-syscollector-dbsync-rsync` branch is not compiling for target `winagent`. Error:
```
    CC win32/ossec-agent.exe
/usr/bin/i686-w64-mingw32-ld: client-agent/receiver.o:receiver.c:(.text+0x72a): undefined reference to `wcom_send'
collect2: error: ld returned 1 exit status
make[1]: *** [Makefile:2027: win32/ossec-agent.exe] Error 1
make[1]: Leaving directory '/home/martin/Documents/Wazuh/code/wazuh/src'
make: *** [Makefile:719: winagent] Error 2
```

- **Actual result:** compilation failed
- **Expected result:** compilation succees
The bug was discovered in: cross platforms compilation testing.
## DoD
- [x] Fix/replace the wcom_send function appropriately
- [x] PR

--- a/src/client-agent/receiver-win.c
+++ b/src/client-agent/receiver-win.c
@@ -156,7 +156,7 @@ void *receiver_thread(__attribute__((unused)) void *none)
 
                 /* syscollector */
                 else if (strncmp(tmp_msg, HC_SYSCOLLECTOR, strlen(HC_SYSCOLLECTOR)) == 0) {
-                    wmcom_sync(tmp_msg);
+                    wmcom_send(tmp_msg);
                     continue;
                 }
 

--- a/src/client-agent/receiver.c
+++ b/src/client-agent/receiver.c
@@ -133,7 +133,7 @@ int receive_msg()
 
             /* syscollector */
             else if (strncmp(tmp_msg, HC_SYSCOLLECTOR, strlen(HC_SYSCOLLECTOR)) == 0) {
-                wcom_send(tmp_msg);
+                wmcom_send(tmp_msg);
                 continue;
             }
 

--- a/src/wazuh_modules/wmcom.c
+++ b/src/wazuh_modules/wmcom.c
@@ -84,19 +84,24 @@ int wmcom_sync(char * buffer) {
     return ret;
 }
 
-#ifndef WIN32
+#ifdef WIN32
+void wmcom_send(char * message)
+{
+    wmcom_sync(message);
+}
+#else
 
-void wcom_send(char * message)
+void wmcom_send(char * message)
 {
     int sock;
     if (sock = OS_ConnectUnixDomain(DEFAULTDIR WM_LOCAL_SOCK, SOCK_STREAM, OS_MAXSTR), sock < 0) {
         switch (errno) {
             case ECONNREFUSED:
-                merror("At wcom_send(): Target wmodules refused connection. The component might be disabled");
+                merror("At wmcom_send(): Target wmodules refused connection. The component might be disabled");
                 break;
 
             default:
-                merror("At wcom_send(): Could not connect to socket wmodules: %s (%d).", strerror(errno), errno);
+                merror("At wmcom_send(): Could not connect to socket wmodules: %s (%d).", strerror(errno), errno);
         }
     }
     else

--- a/src/wazuh_modules/wmodules.h
+++ b/src/wazuh_modules/wmodules.h
@@ -177,8 +177,8 @@ void * wmcom_main(void * arg);
  *
  * @param message Payload.
  */
-void wcom_send(char * message);
 #endif
+void wmcom_send(char * message);
 size_t wmcom_dispatch(char * command, char ** output);
 size_t wmcom_getconfig(const char * section, char ** output);
 int wmcom_sync(char * buffer);

